src/lazy/any_encoding.rs

```diff
@@ -62,23 +72,99 @@ impl LazyDecoder for AnyEncoding {
     type SExp<'top> = LazyRawAnySExp<'top>;
     type List<'top> = LazyRawAnyList<'top>;
     type Struct<'top> = LazyRawAnyStruct<'top>;
+    type FieldName<'top> = LazyRawAnyFieldName<'top>;
     type AnnotationsIterator<'top> = RawAnyAnnotationsIterator<'top>;
-    type EExpression<'top> = LazyRawAnyEExpression<'top>;
+    type EExp<'top> = LazyRawAnyEExpression<'top>;
+    type VersionMarker<'top> = LazyRawAnyVersionMarker<'top>;
+}
+
+#[derive(Debug, Copy, Clone)]
+pub struct LazyRawAnyVersionMarker<'top> {
+    encoding: LazyRawAnyVersionMarkerKind<'top>,
 }
+
+#[derive(Debug, Copy, Clone)]
+pub enum LazyRawAnyVersionMarkerKind<'top> {
+    Text_1_0(LazyRawTextVersionMarker_1_0<'top>),
+    Binary_1_0(LazyRawBinaryVersionMarker_1_0<'top>),
+    Text_1_1(LazyRawTextVersionMarker_1_1<'top>),
+    Binary_1_1(LazyRawBinaryVersionMarker_1_1<'top>),
+}
+
+impl<'top> HasSpan<'top> for LazyRawAnyVersionMarker<'top> {
+    fn span(&self) -> Span<'top> {
+        use LazyRawAnyVersionMarkerKind::*;
+        match self.encoding {
+            Text_1_0(marker) => marker.span(),
+            Binary_1_0(marker) => marker.span(),
+            Text_1_1(marker) => marker.span(),
+            Binary_1_1(marker) => marker.span(),
+        }
```

Comment:
🗺️ This file (`LazyRawAny_____` types) is entirely enum dispatch. A while ago I looked to see if I could use the `enum_dispatch` crate to eliminate all the repetition, but the use of lifetimes was an obstacle. I have not checked recently to see if that limitation has been overcome.